lnrpc-ideal/invoicesrpc/invoices.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+import "google/api/annotations.proto";
+import "rpc.proto";
+
+package invoicesrpc;
+
+option go_package = "github.com/lightningnetwork/lnd/lnrpc/invoicesrpc";
+
+// Invoices is a service that can be used to create, accept, settle and cancel
+// invoices.
+service Invoices {
+    /**
+    SubscribeSingleInvoice returns a uni-directional stream (server -> client)
+    to notify the client of state transitions of the specified invoice.
+    Initially the current invoice state is always sent out.
+    */
+    rpc SubscribeSingleInvoice (SubscribeSingleInvoiceRequest)
```

Comment:
We are missing a `LookupInvoice` call, but this one can double as that because it immediately sends out the current invoice state.

Reply:
Assuming a world where we've worked out streaming for REST, which seems to be on the horizon.